examples/typescript/local_node.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable import/no-commonjs */
+/* eslint-disable import/extensions */
+
+/**
+ * Example to show how one can require the CLI module and use it
+ * to run cli commands
+ */
+
+const cli = require("@aptos-labs/ts-sdk/dist/common/cli/index.js");
+
+// Run local node
+async function runLocalNode() {
+  const localNode = new cli.LocalNode();
+  await localNode.run();
+}
+
+// initialize current directory for Aptos
+async function init() {
+  const move = new cli.Move();
+
+  await move.init({
+    network: "local",
+    profile: "default",
+  });
+}
+
+// compile a package
+async function compile() {
+  const move = new cli.Move();
+
+  await move.compile({
+    packageDirectoryPath: "move/moonCoin",
+    namedAddresses: {
+      MoonCoin: "0x123",
+    },
+  });
+}
+
+// run Move unit tests for a package
+async function tests() {
+  const move = new cli.Move();
+
+  await move.test({
+    packageDirectoryPath: "move/moonCoin",
+    namedAddresses: {
+      MoonCoin: "0x123",
+    },
+  });
+}
+
+// publish the Move package to the publisher's account
+async function publish() {
+  const move = new cli.Move();
+
+  await move.publish({
+    packageDirectoryPath: "move/moonCoin",
+    namedAddresses: {
+      // please replace the address with the actual address
+      MoonCoin: "0x123",
+    },
+  });
+}
+
+// create a new object and publish a Move package to it
+async function createObjectAndPublishPackage() {
+  const move = new cli.Move();
+
+  await move.createObjectAndPublishPackage({
```

Comment:
what is the output we are getting from this command? is this the object address?

Reply:
yes, in this format

```sh
➜  sh_scripts ./deploy_nft_template.sh
##### Deploy module under a new object #####
Compiling, may take a little while to download git dependencies...
UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-core.git
UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-core.git
UPDATING GIT DEPENDENCY https://github.com/aptos-labs/token-minter.git
UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-core.git
UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-core.git
UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-core.git
INCLUDING DEPENDENCY AptosFramework
INCLUDING DEPENDENCY AptosStdlib
INCLUDING DEPENDENCY AptosToken
INCLUDING DEPENDENCY AptosTokenObjects
INCLUDING DEPENDENCY MoveStdlib
INCLUDING DEPENDENCY TokenMinter
BUILDING digital_asset_launchpad
package size 12447 bytes
Code was successfully deployed to object address 0x2608d20d0712a847296ccf4a768e63df351dc53153fba43458ad166048a8a1da.
{
  "Result": "Success"
}
```

Reply:
I used to have [this shell](https://github.com/aptos-labs/move-by-examples/blob/main/fungible-asset-launchpad/aptos/move/sh_scripts/deploy.sh#L13-L22) script to extract it and save it to txt, will replicate it in cad

Reply:
could we return the object address? and then, in a project (cad for example), use this output to run upgradeObjectPackage command?

Reply:
If this is the recommended way to publish a move module - why not implement it in the publish function? so that would be the default way

Reply:
my reason is to keep the ts function name consistent to cli name.

Reply:
I understand. Tho I think it is an option for us to "update" the cli API and make it more consistent with the standard and expectations we have from our dev users and provide a better dev exp. Also, on CAD we still want to use `npm run move:publish` to publish the module instead  of using a confusing name like `npm run move:createObjectAndPublishPackage` so names will still be different